File: backend/Admin/src/main/java/com/example/entities/Rating.java
package com.example.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Entity
@Getter
@Setter
@Table(name = "ratings")
public class Rating {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int ratingId;

    @ManyToOne
    @JoinColumn(name = "mess_id")
    private Mess mess;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private int rating;

    private String comments;

}

File: backend/Customer/Customer/Models/Rating.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class Rating
{
    public int RatingId { get; set; }

    public int MessId { get; set; }

    public int UserId { get; set; }

    public int Rating1 { get; set; }

    public string? Comments { get; set; }

    public virtual Mess Mess { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
